discovery-infra/test_infra/junit_report/junit_test_suite.py

```diff
@@ -0,0 +1,54 @@
+import os
+from typing import Callable, Union, Type
+
+import decorator
+from junit_xml import TestSuite, to_xml_report_string
+
+from .junit_test_case import JunitTestCase
+
+
+class JunitTestSuite:
+    _report_dir: str
+    _func: Union[Callable, None]
+    _suite: Union[TestSuite, None]
+    _name = str
+
+    def __init__(self, report_dir="./"):
+        self._report_dir = report_dir
+        self._func = None
+        self._suite = None
+        self._name = None
+
+    def __call__(self, function: Callable):
+        self._func = function
+        self._name = function.__name__
+
+        def wrapper(_, obj: Type[object], *args, **kwargs):
```

Comment:
What's the first parameter being ignored?
What's `obj`? In case it could be anything, you should use `Any` if that's a class, you should use `Class`. 
I think you don't expect here an `int`

Reply:
First parameter is the same function on line 23, obj is the class that uses the decorator

Reply:
I think you can keep the first variable as `function` instead of using `_`, although it's unused.
Change the typehint for the obj for class